sdk/messaging_servicebus/src/service_bus/topic_client.rs

```diff
@@ -47,6 +47,9 @@ impl TopicClient {
         P: Into<String>,
         K: Into<Secret>,
     {
+        // NOTE: This is to account for the azure_core::auth::hmac_sha256 assumption
+        // that the key needs to be base64 decoded.
+        let signing_key = azure_core::base64::encode(signing_key.into().secret());
```

Comment:
How feasible is it just to change the `TopicClient` to accept the raw string? Seems a waste to encode it just to have it decoded.

Reply:
I'm not sure it's feasible, but I think it's more important to correct this regression first.
We can think about a redesign afterwards.
Even if it's not ideal, doing this base64 encoding/decoding is probably not significant in terms of performance compared to the http calls made by the topic client. And the base64 thing is only done once when the client is created.